docs/adr/0004-withdraw-expiry.rst

```diff
@@ -0,0 +1,91 @@
+Withdraw & Expiry
+=================
+
+**Status**: proposed
+**Deciders**: @konradkonrad @palango @hackaugusto @rakanalh
+**Date**: 2019-07-03
+
+Context and Problem Statement
+=============================
+
+Withdraw functionality was introduced into Raiden to enable users to withdraw funds from their open channels into their accounts.
+The implemented flow works as follows:
+
+1. Node (A) initiates the withdraw by dispatching `ActionInitWithdraw` with a monitonic total withdraw value.
+2. Node (A) sends a signed `WithdrawRequest` message to partner to agree on the new total withdraw.
+3. Node (B) receives the withdraw request, validates it and replies with a signed `WithdrawConfirmation` message to node (A).
+4. Node (A) receives the withdraw confirmation, validates it and uses signatures from step (1) and step(2) to send an on-chain transaction
+   to withdraw the funds.
+
+Here, a problem arises when node (B) is unavailable or is not cooperative. In this case, node (A) would have sent a withdraw request,
+locked part or all of it's deposited amount for withdraw and no confirmation was received from node (B). This will lead to node (A) not being able
+to use the locked funds for withdraw until node (B) replies. In the case of node (B) not being cooperative on this withdraw request, node (A)'s
+locked withdraw funds are unavailable indefinitely until the channel is closed.
+
+Considered Options
+==================
+
+* **Option 1**: Node (A) could close the channel with node (B). In this case, node (A) gets back the funds it originally locked for withdraw.
```

Comment:
I like approach 1. It's simple, easy to implement and does not complicate our protocol further with more messages.

Reply:
We had a discussion about this and ended up deciding on introducing a `WithdrawExpired`.

Reply:
Hmm that's not good enough :smile: 

The whole point of an ADR is to introduce with proper arguments both options and explain why one is better than the other and why it was chosen. If it's already decided then why make the ADR in the first place?

By reading the document I don't see the advantages of Option 2. I just see another message complicating the protocol further (Do I even need to mention `LockExpired` and the legion of bugs it introduced?) whereas a much simpler and easier solution exists.

From an ADR I would expect an explanation of why an option was chosen. Not "we discussed it and ended up going for it". Remember it's for people who were not there during the discussion, and also your future selves who will not remember what was discussed!

Can you maybe elaborate on that in the document? Then I (and others in the future) may be able to see the advantage of the `WithdrawExpired` message.

Reply:
@LefterisJP added the reasoning behind going for Option 2. I hope this satisfies your request.